.github/workflows/code-checks.yml

```diff
@@ -100,6 +91,12 @@ jobs:
       run: ci/code_checks.sh typing
       if: ${{ steps.build.outcome == 'success' }}
 
+    - uses: jakebailey/pyright-action@v1
+      with:
+        # note: keep version in sync with .pre-commit-config.yaml
```

Comment:
Curious, why do we need this in pre-commit-config and here i.e. can this just be run in pre-commit?

Reply:
pyright is only run by pre-commit during the `manual` stage. The idea is to give developers an easy way to run pyright without having to set it up (it depends on node). The CI doesn't run the `manual` stage.

I think the reason to keep mypy and pyright separate was that otherwise all the typing packages would need to be installed for the pre-commit hook. Maybe there was a better reason :)

Reply:
the reason for not running pyright&mypy (by default) in pre-commit is that outdated/missing packages (e.g., numpy) would cause pyright&mypy to fail. We often have issues between numba requiring an older version of numpy but for typing we want the latest numpy.

Reply:
@Dr-Irv I think you set the pyright github action for https://github.com/pandas-dev/pandas-stubs up. How did you manage that pyright finds the virtual env? Running pyright directly (instead of using the github action) seemed to have no issue finding the virtual env.

Reply:
Make sense for typing checks in pre-commit to not necessarily run on every commit and therefore in `manual` mode. Is it possible to have the pre-commit job run these pyright checks then in the CI? My motivation is to see if there can be one clear way to run these checks.

Reply:
It would be great to only use pre-commit and make mypy&pyright optional!

I think one issue with mypy+pre-commit is that pre-commit isolates the python environment (can probably be avoided by `language: system`)? Pyright doesn't seem to be affected by that since pre-commit doesn't seem to isolate non-python programs.

Reply:
> @Dr-Irv I think you set the pyright github action for https://github.com/pandas-dev/pandas-stubs up. How did you manage that pyright finds the virtual env? Running pyright directly (instead of using the github action) seemed to have no issue finding the virtual env.

I just copied the action from what Microsoft had set up in https://github.com/microsoft/python-type-stubs/blob/main/.github/workflows/test.yml .  I think since it is a sequence of steps, the virtual env is automatically set up from the previous steps.